operator/src/test/java/org/keycloak/operator/testsuite/integration/TracingDeploymentTest.java

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright 2024 Red Hat, Inc. and/or its affiliates
+ * and other contributors as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.keycloak.operator.testsuite.integration;
+
+import io.quarkus.test.junit.QuarkusTest;
+import org.awaitility.Awaitility;
+import org.junit.jupiter.api.Test;
+import org.keycloak.operator.Constants;
+import org.keycloak.operator.crds.v2alpha1.deployment.Keycloak;
+import org.keycloak.operator.crds.v2alpha1.deployment.ValueOrSecret;
+import org.keycloak.operator.crds.v2alpha1.deployment.spec.FeatureSpec;
+
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.keycloak.operator.controllers.KeycloakDeploymentDependentResource.KC_TRACING_RESOURCE_ATTRIBUTES;
+import static org.keycloak.operator.controllers.KeycloakDeploymentDependentResource.KC_TRACING_SERVICE_NAME;
+import static org.keycloak.operator.testsuite.utils.K8sUtils.deployKeycloak;
+
+@QuarkusTest
+public class TracingDeploymentTest extends BaseOperatorTest {
+
+    public static Keycloak getTestKeycloakDeployment(boolean disableProbes) {
+        var kc = BaseOperatorTest.getTestKeycloakDeployment(disableProbes);
+        kc.getSpec().setStartOptimized(false);
```

Comment:
Looks good. This may have gotten disabled prior to recent change to make it a hard error to start optimized with a changed build time option - that would have made it in the pod log what was happening.

However I'm wondering why it ever worked. There should have been a warning that tracing-enabled was ignored.

Reply:
I think we should handle pod logs better if some issue occurs - still space for improvements. As sometimes it's really hard to understand what exactly happened and why the test failed. 

> However I'm wondering why it ever worked.

The thing is in the test case, we're just checking the env vars propagated to the pod. And when we have disabled probes for the test deployment, the test does not even wait to properly start the server up. 

I'll enable these probes to verify the pod started correctly.